### DISCUSSION TITLE
Add Gcode wipe nozzle over x layer

### DISCUSSION THREAD

**Discussion by erwann17** [OrcaSlicer/OrcaSlicer#7256] [Q&A]
Hello ,
I would like to automatically integrate a Gcode Wipe nozzle every 5 layers for each cutting but I cannot find a solution to integrate this into orca slicer.
could you help yourself

**Comment by LazaroFilm** [comment]
SOLUTION:
I created a macro in Klipper for this. 

- In OrcaSlicer, add `_WIPE_AT_LAYER WIPE_EVERY=10 LAYER={layer_num}` to the end of your **Layer Change G-code** block. This calls the Klipper macro with the variable of the current layer. If you don't want to wipe every x layers, set `WIPE_EVERY=0`:
```
[gcode_macro _WIPE_AT_LAYER]
gcode:
  {% set LAYER = params.LAYER | default(0) | int %}           
  {% set WIPE_EVERY = params.WIPE_EVERY | default(0) | int %} 
  {% if WIPE_EVERY > 0 and LAYER % WIPE_EVERY == 0 %}
    WIPE
  {% endif %}
```
Make sure you have a `WIPE` macro defined in your configuration.
